### ISSUE TITLE
Bad request in CONNECT with Body of 0 size

### ISSUE THREAD

**Issue by ra-at-diladele-com** [diladele/webproxy#218]
```
Connecting to any https site it blocks by "Bad request"
```

**Comment by ra-at-diladele-com** [comment]
```
[2021-Apr-09 10:07:14] 9168 | listener.cpp(183) | accept_next | context 1 constructed
[2021-Apr-09 10:07:20] 9200 | accepted_plain_connection.cpp(32) | accepted_plain_connection | constructed
[2021-Apr-09 10:07:20] 9200 | accepted_plain_connection.cpp(63) | process | expected request is initially proxy_style
[2021-Apr-09 10:07:20] 9200 | accepted_plain_connection.cpp(64) | process | selected policy default
[2021-Apr-09 10:07:20] 9200 | waiting_for_request.cpp(40) | waiting_for_request | constructed
[2021-Apr-09 10:07:20] 8724 | waiting_for_request.cpp(91) | handle_read_from_client | client >>> CONNECT checkappexec.microsoft.com:443 HTTP/1.0

Host: checkappexec.microsoft.com:443

Content-Length: 0

Proxy-Connection: Keep-Alive

Pragma: no-cache




[2021-Apr-09 10:07:20] 8724 | waiting_for_request.cpp(110) | handle_read_from_client | created request: 0
[2021-Apr-09 10:07:20] 8724 | scanning_request_headers.cpp(26) | scanning_request_headers | constructed
[2021-Apr-09 10:07:20] 8724 | scanning_request_headers.cpp(35) | scan_headers | starting scan of headers
[2021-Apr-09 10:07:20] 8724 | scanning_request_headers.cpp(92) | handle_request_headers | handle_request_headers
[2021-Apr-09 10:07:20] 8724 | tunnel_processor.cpp(22) | tunnel_processor | constructed
[2021-Apr-09 10:07:20] 8724 | request_failer.hpp(18) | fail_plain_request | fail_plain_request
[2021-Apr-09 10:07:20] 8724 | replying_with_plain_error.hpp(37) | replying_with_plain_error | constructed
[2021-Apr-09 10:07:20] 8724 | replying_with_plain_error.hpp(72) | reply_and_terminate | client <<< HTTP/1.1 400 Bad Request

X-WebProxy-Version: 0.12

Content-Type: text/html; charset=utf-8

Content-Length: 6709

Connection: close



<!DOCTYPE HTML>
<html>
    <head>
        <meta charset="utf-8"/>
        <title>400 Bad Request</title>
       
    </head>
    <body onLoad="checkIfTopMostWindow()">
        <div id="wrapper">
            <h1>:(</h1>
            <h2>400 Bad Request</h2>    

            

<p>Web Filtering Proxy running on server <strong>DESKTOP-3DERQBP</strong> was not able forward the client request to the origin server. The request seems to be malformed or incorrect.</p>

...                  

                    <tr><td class="info_title">Additional Info</td><td class="info_content">Browser asked the proxy to establish the CONNECT tunnel to remote origin server. Unfortunately the request contains 'fixed_body' of '0' bytes. According to RFC 7230 (3.3) CONNECT tunnel request shall not contain a body. This tunnel cannot be established, connection between the browser and the proxy will now be terminated.</td></tr>
...

[2021-Apr-09 10:07:20] 9200 | replying_with_plain_error.hpp(117) | handle_written_to_client | last write to client completed
[2021-Apr-09 10:07:20] 9200 | terminating.hpp(23) | terminating | constructed
[2021-Apr-09 10:07:20] 9200 | context.cpp(43) | ~context | context 1 destroyed
```

**Comment by ra-at-diladele-com** [comment]
Fixed in 0.14; need to test.
